dbt/models/reporting/reporting.ratio_stats.py

```diff
@@ -4,14 +4,462 @@
+    :param kwargs:
+        Arguments passed on to ``fun``.
+    :type fun: function
+    :type nboot: int
+    :type alpha: float
+    :type kwargs: numeric
+
+    .. note::
+       Input function should require 1 argument or be ``assesspy.prd()``.
+
+    :return:
+        A two-long list of floats containing the bootstrapped confidence
+        interval of the input vector(s).
+    :rtype: list[float]
+
+    :Example:
+
+    .. code-block:: python
+
+        # Calculate PRD confidence interval:
+        import assesspy as ap
+
+        ap.boot_ci(
+            ap.prd,
+            assessed = ap.ratios_sample().assessed,
+            sale_price = ap.ratios_sample().sale_price,
+            nboot = 100
+            )
+    """
+
+    # Make sure prd is passed arguments in correct order
+    if fun.__name__ == "prd" and set(["assessed", "sale_price"]).issubset(
+        kwargs.keys()
+    ):
+        kwargs = (kwargs["assessed"], kwargs["sale_price"])
+    elif fun.__name__ == "prd" and not set(
+        ["assessed", "sale_price"]
+    ).issubset(kwargs.keys()):
+        raise Exception(
+            "PRD function expects argurments 'assessed' and 'sale_price'."
+        )
+    else:
+        kwargs = tuple(kwargs.values())
+
+    check_inputs(kwargs)  # Input checking and error handling
+
+    num_kwargs = len(kwargs)
+    kwargs = pd.DataFrame(kwargs).T
+    n = len(kwargs)
+
+    # Check that the input function returns a numeric vector
+    out = (
+        fun(kwargs.iloc[:, 0])
+        if num_kwargs < 2
+        else fun(kwargs.iloc[:, 0], kwargs.iloc[:, 1])
+    )
+    if not is_numeric_dtype(out):
+        raise Exception("Input function outputs non-numeric datatype.")
+
+    data_array = kwargs.to_numpy()
+
+    def bootstrap_worker(
+        data_array, fun, num_kwargs, n, nboot, start, end, result_queue
+    ):
+        ests = []
+        for _ in range(start, end):
+            sample_indices = np.random.choice(
+                data_array.shape[0], size=n, replace=True
+            )
+            sample_array = data_array[sample_indices]
+            if fun.__name__ == "cod" or num_kwargs == 1:
+                ests.append(fun(sample_array[:, 0]))
+            elif fun.__name__ == "prd":
+                ests.append(fun(sample_array[:, 0], sample_array[:, 1]))
+            else:
+                raise Exception(
+                    "Input function should require 1 argument or be assesspy.prd."  # noqa
+                )
+        result_queue.put(ests)
+
+    def parallel_bootstrap(
+        data_array, fun, num_kwargs, n, nboot, num_processes=4
+    ):
+        processes = []
+        result_queue = mp.Queue()
+        chunk_size = nboot // num_processes
+
+        for i in range(num_processes):
+            start = i * chunk_size
+            end = start + chunk_size if i < num_processes - 1 else nboot
+            p = mp.Process(
+                target=bootstrap_worker,
```

Comment:
This function allocates the size of each process, and divides the `ets` calculation into `n_processes`